src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -90,16 +92,20 @@ def register_callback(
 
         Raises
         ------
-        RuntimeError
+        InvalidArgument
             If event name is not valid.
+        DisallowedValuesError
+            If an argument value not in allowed values.
         """
         if event_name is None or call_back is None:
-            raise RuntimeError(
+            raise InvalidArgument(
                 "Please provide compulsory arguments : 'event_name' and 'call_back'"
             )
 
         if event_name not in self.events_list:
-            raise RuntimeError(f"{event_name} is not a valid event.")
+            raise DisallowedValuesError(
+                allowed_name_error_message("event-name", event_name, self.events_list)
```

Comment:
This is similar to all those other issues I commented on. In this case, the caller doesn't need to tell the exception type how to build the generalised error message for that type.

Reply:
Right. Thank you.